src/FakeItEasy/ArgumentEqualityComparer.cs

```diff
@@ -0,0 +1,43 @@
+namespace FakeItEasy;
+
+using System;
+
+/// <summary>
+/// Provides custom equality comparison for call arguments for parameters of type T.
+/// </summary>
+/// <typeparam name="T">Tye type of parameter to compare.</typeparam>
+public abstract class ArgumentEqualityComparer<T> : IArgumentEqualityComparer
+{
+    /// <summary>
+    /// Gets the priority of the argument equality comparer. When multiple comparers that apply to the same type are registered,
+    /// the one with the highest priority value is used.
+    /// </summary>
+    public virtual Priority Priority => Priority.Default;
+
+    /// <summary>
+    /// Whether or not this object can compare instances of <paramref name="type"/> for equality.
+    /// </summary>
+    /// <param name="type">The type of the objects to compare.</param>
+    /// <returns>
+    /// <c>true</c> if this object can compare instances of <paramref name="type"/>. Otherwise <c>false</c>.
+    /// </returns>
+    public virtual bool CanCompare(Type type) => type == typeof(T);
+
+    /// <summary>
+    /// Indicates whether <paramref name="expectedValue"/> and <paramref name="argumentValue"/> are considered equal.
+    /// </summary>
+    /// <param name="expectedValue">The first object to compare.</param>
+    /// <param name="argumentValue">The second object to compare.</param>
+    /// <returns><c>true</c> if the objects are considered equal. Otherwise <c>false</c>.</returns>
+    /// <remarks>This method cannot be overriden. Override the <see cref="AreEqual(T, T)"/> method instead.</remarks>

```

Comment:
great remark!

Reply:
But you missed the typo ^^
"overriden" => "overridden"

Reply:
HA!